### ISSUE TITLE
exists on disk, but not in 'origin/master'  error with absolute paths

### ISSUE THREAD

**Issue by ennozdd** [demisto/demisto-sdk#2318]
```
$ demisto-sdk validate -i Packs/mysweetpack/
You are using demisto-sdk 1.7.4.

Running in a private repository

================= Validating pack /home/ennozdd/ennozdd/Packs/mysweetpack =================

Validating /home/ennozdd/ennozdd/Packs/mysweetpack unique pack files
Got an error while trying to connect to git - Cmd('git') failed due to: exit code(128)
  cmdline: git show origin/master:/home/ennozdd/ennozdd/Packs/mysweetpack/pack_metadata.json
    stderr: 'fatal: Path '/home/ennozdd/ennozdd/Packs/mysweetpack/pack_metadata.json' exists on disk, but not in 'origin/master'.'
```

Even though the pack's location is given as a relative path, demisto-sdk chooses to go with its absolute path. Absolute paths seem to not work with git show it should use the relative form from the repo root. 

Example:
$ git show origin/master:/home/ennozdd/ennozdd/Packs/mysweetpack/pack_metadata.json
fatal: Path '/home/ennozdd/ennozdd/Packs/mysweetpack/pack_metadata.json' exists on disk, but not in 'origin/master'.
$ git show origin/master:Packs/mysweetpack/pack_metadata.json

https://stackoverflow.com/a/65783920

**Comment by dorschw** [comment]
Hi @ennozdd, thanks for reporting this issue, this is tracked internally as [CIAC-4000](https://jira-hq.paloaltonetworks.local/browse/CIAC-4000)

**Comment by upstart-swiss** [comment]
Also encountering this here and in CI. Any recommendation of how to better run the command to not have this error?

**Comment by DinaMeylakh** [comment]
@ennozdd @upstart-swiss This is actually supposed to be handled by the line
```
relative_file_path = os.path.relpath(full_file_path, self.git_path())
``` 
where the function is defined by

```    
def git_path(self) -> str:
        git_path = self.repo.git.rev_parse('--show-toplevel')
        return git_path.replace('\n', '')
```

@ennozdd Could you run `git rev-parse --show-toplevel` and see if it returns '/home/ennozdd/ennozdd/' as expected?

Also, are you getting a message like the following?
```
Could not get local remote file because of: ...
Searching the remote file content with the API.
```
If so, it will be helpful if you could paste the message here.

@upstart-swiss My wild guess is that the repo is not configured correctly at the time of running the command, but more information is required for a proper solution.

**Comment by upstart-swiss** [comment]
Here's the relevant part of the build from our Github Actions workflow:
```
  build:
    needs: [determine_changes]
    if: ${{ needs.determine_changes.outputs.Packs == 'true' }}
    runs-on: ubuntu-latest
    permissions:
      id-token: write
      contents: read
      pull-requests: write
      checks: write
      actions: write
    env:
      # secrets
      AWS_ACCOUNT_ID: ${{ github.ref_name == 'main' && secrets.AWS_ACCOUNT_ID_PRODUCTION || secrets.AWS_ACCOUNT_ID_PLAYGROUND }}
      # default vars
      AWS_REGION: us-east-1
      AWS_WEB_IDENTITY_TOKEN_FILE: /tmp/oidcawscreds
      BRANCH_NAME: ${{ github.ref_name }}
      DEFAULT_BRANCH: ${{ github.event.repository.default_branch }}
      DEMISTO_README_VALIDATION: false
      DEMISTO_SDK_GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
    steps:
      - name: XSOAR CI/CD master checkout (Main)
        if: github.event_name == 'push'
        uses: actions/checkout@v2
        with:
          path: repository
          fetch-depth: 0
      - name: XSOAR CI/CD master checkout (PR)
        if: github.event_name == 'pull_request'
        uses: actions/checkout@v2
        with:
          path: repository
          ref: ${{ github.event.pull_request.head.ref }}
          fetch-depth: 0
      - name: Content checkout
        uses: actions/checkout@v2
        with:
          repository: demisto/content
          path: content
      - name: Setup Python
        uses: actions/setup-python@v2
        with:
          python-version: 3.8.5
      - name: Install pip and requirements
        run: |
          pip install --upgrade pip
          pip install -r ${GITHUB_WORKSPACE}/repository/requirements.txt
      - name: Prepare Environment
        run: |
          echo "PATH=/home/runner/.local/bin:$PATH" >> $GITHUB_ENV
          echo "PYTHONPATH=${GITHUB_WORKSPACE}/content:${GITHUB_WORKSPACE}:${PYTHONPATH}" >> $GITHUB_ENV
          echo "ARTIFACTS_FOLDER=${GITHUB_WORKSPACE}/repository/content/artifacts" >> $GITHUB_ENV
          PACKS_CHANGED=$(python $GITHUB_WORKSPACE/repository/build_related_scripts/get_modified_packs.py --repo_path $GITHUB_WORKSPACE/repository/ --prev-ver ${DEFAULT_BRANCH})
          echo "PACKS_CHANGED=${PACKS_CHANGED}" >> $GITHUB_ENV
      - name: Prepare Venv
        run: |
          cd $GITHUB_WORKSPACE/content/
          if [ ! -e "venv" ]; then
            echo "installing venv"
            cp $GITHUB_WORKSPACE/repository/dev-requirements-py3.txt $GITHUB_WORKSPACE/content/.
            NO_HOOKS=1 SETUP_PY2=no $GITHUB_WORKSPACE/repository/.hooks/bootstrap
            source ./venv/bin/activate
            pip3 install -r ${GITHUB_WORKSPACE}/repository/requirements.txt
            demisto-sdk --version
          else
            echo "venv exists (from cache). activating"
            source ./venv/bin/activate
          fi
          # Install node packages
          npm install .
          git config diff.renameLimit 6000
          pip3 install -U demisto-sdk
          echo "========== Build Parameters =========="
          python3 --version
          demisto-sdk --version
      - name: Create ID Set
        run: |
          if [ $PACKS_CHANGED ]; then
            cd $GITHUB_WORKSPACE/repository/
            source $GITHUB_WORKSPACE/content/venv/bin/activate
            # Download content's
            gsutil cp gs://marketplace-dist/content/id_set.json ./Tests/content-id_set.json
            # Create repository id-set
            demisto-sdk create-id-set
            # Merge id-sets and copy to artifacts folder
            demisto-sdk merge-id-sets -i1 ./Tests/id_set.json -i2 ./Tests/content-id_set.json -o ./Tests/id_set.json
            mkdir -p ${ARTIFACTS_FOLDER}
            cp "./Tests/id_set.json" "$ARTIFACTS_FOLDER/id_set.json"
          else
            echo "No packs has changed, skipping step."
          fi
      - name: Validate Files and Yaml
        id: demisto_validate
        if: always()
        continue-on-error: true
        run: |
          source $GITHUB_WORKSPACE/content/venv/bin/activate
          export NODE_PATH=$GITHUB_WORKSPACE/content/node_modules
          cd $GITHUB_WORKSPACE/repository
          mkdir -p $GITHUB_WORKSPACE/repository/Tests/Marketplace
          cp $GITHUB_WORKSPACE/content/Tests/Marketplace/approved_tags.json $GITHUB_WORKSPACE/repository/Tests/Marketplace/
          cp $GITHUB_WORKSPACE/content/Tests/Marketplace/approved_usecases.json $GITHUB_WORKSPACE/repository/Tests/Marketplace/
          # Run validate on all changed files
          # You can choose what validations to use in the .demisto-sdk-conf file
          demisto-sdk validate --quiet-bc-validation --no-conf-json --allow-skipped --skip-pack-dependencies --prev-ver ${DEFAULT_BRANCH} | tee validate_results.txt
```

**Comment by DinaMeylakh** [comment]
Hi @upstart-swiss 
Thanks for the details. It is not clear to me that you have a repo configured in your build. 
Could you add the command `git rev-parse --show-toplevel` before the `demisto-sdk validate` and make sure it returns the path to the directory which contains your `Packs` directory?

**Comment by ennozdd** [comment]
@DinaMeylakh Hey, I ran the command `git rev-parse --show-toplevel` and it gave me the correct location.
/home/ennozdd/ennozdd/

**Comment by DinaMeylakh** [comment]
Hi @ennozdd, Great catch & thanks for the contribution! I've added a review to ensure XSOAR's development conventions in the demisto-sdk repo. If you find it a little much and you'd prefer that I fix the bug, let me know.

**Comment by ennozdd** [comment]
@DinaMeylakh I prefer that you fix the bug :) thank you :)

**Comment by DinaMeylakh** [comment]
@ennozdd On it :)

**Comment by DinaMeylakh** [comment]
@ennozdd @upstart-swiss 
The fix was merged [here](https://github.com/demisto/demisto-sdk/pull/2558)
And should be available in the next demisto-sdk release (1.8.1 and on)

**Comment by ennozdd** [comment]
Thank you. 1.8.1 is released with the fix so I think I can close the issue now
